ocs_ci/utility/utils.py

```diff
@@ -2356,16 +2432,35 @@ def run_ceph_health_cmd(namespace):
 
     """
     # Import here to avoid circular loop
-    from ocs_ci.ocs.resources.pod import get_ceph_tools_pod
+    if namespace == constants.OPENSHIFT_STORAGE_EXTENDED_NAMESPACE:
+        from ocs_ci.utility.connection import Connection
+        from ocs_ci.deployment.helpers.external_cluster_helpers import (
+            get_external_cluster_client,
+        )
 
-    try:
-        ct_pod = get_ceph_tools_pod(namespace=namespace)
-    except (AssertionError, CephToolBoxNotFoundException) as ex:
-        raise CommandFailed(ex)
+        host, user, password, ssh_key = get_external_cluster_client()
+        if not password and ssh_key:
+            connection_to_cephcluster = Connection(
+                host=host, user=user, private_key=ssh_key
+            )
+        elif password:
+            connection_to_cephcluster = Connection(
+                host=host, user=user, password=password
+            )
```

Comment:
Connection itself handle already the ssh_key with higher priority as you can see here:
https://github.com/shyRozen/ocs-ci/blob/9ca9c4e26e8fd361962d410d2e297f3bfeddb609/ocs_ci/utility/connection.py#L66

```suggestion
        connection_to_cephcluster = Connection(
                host=host, user=user, password=password, private_key=ssh_key
         )
```